addons/web/static/tests/views/form_tests.js

```diff
@@ -4854,6 +4854,70 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('correct amount of buttons', function (assert) {
+        assert.expect(7);
+
+        var buttons = '';
+
+        for(var i = 0; i < 7; i++) {
+            buttons += '<button type="object" class="oe_stat_button" icon="fa-check-square">' +
+                '<field name="bar"/>' +
+            '</button>';
+        }
+
+        var createFormWithDeviceSizeClass = function (self, size_class) {
+            return createView({
+                View: FormView,
+                model: 'partner',
+                data: self.data,
+                arch: '<form>' +
+                    '<div name="button_box" class="oe_button_box">'
+                        + buttons +
+                    '</div>' +
+                '</form>',
+                res_id: 2,
+                config: {
+                    device: {size_class: size_class},
+                },
+            });
+        }
+
+        var countStatButtons = function (form) {
```

Comment:
assert.containsN for all this function's calls
Jquery selectors support :not()
https://github.com/odoo/odoo/blob/13.0/addons/web/static/tests/helpers/qunit_asserts.js#L95